### PR TITLE
Ssl support

### PR DESCRIPTION
**Pull request by ElvinEfendi**
fixes https://github.com/Shopify/traffic/issues/1101

The PR is the continuation/completion of https://github.com/openresty/lua-resty-upstream-healthcheck/pull/5.

In this PR the comments to original PR are addressed(mainly session reuse and tests) and the code is rebased with the latest upstream.

One significant modification to the original PR is that now, the health check will fail if `ssl_verify = true` and the certificate verification fails.

/r @charlescng @klautcomputing @csfrancis 
/cc @Shopify/traffic

### CODE REVIEW

**Review by klautcomputing — APPROVED** [review]
Should we maybe switch the `no plan` for the tests back to plan and set it to `108` which is the number of tests we are executing here.

Also really smart to test the session reuse that way!

File: lib/resty/upstream/healthcheck.lua [hunk 1]
```diff
@@ -230,16 +230,26 @@ local function check_peer(ctx, id, peer, is_backup)
...
-        if not peer.down then
-            errlog("failed to connect to ", name, ": ", err)
+        return peer_error(ctx, is_backup, id, peer, "failed to connect to ", name, ": ", err)
+    end
+
+    if ctx.type == "https" then
+        session, err = sock:sslhandshake(ctx.session, name, ctx.ssl_verify)
+        if not session then
+            peer_error(ctx, is_backup, id, peer,
+                       "failed to do SSL handshake: ", name, ": ", err)
```
Comment (charlescng) [inline lib/resty/upstream/healthcheck.lua:240]: `SSL handshake failed`? `failed to do SSL handshake` sounds to me that it failed to initiate the handshake vs the handshake failed.

**Review by charlescng — APPROVED** [review]
Looks good!

**Comment by gentle-king** [comment]
Any update? seems not yet released.

and one thing, if upstream server verify cert, so, how could health check take a cert when do handshake?

**Pull request merged**